flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -187,6 +191,13 @@ public YarnClusterDescriptor(
 		return files.isEmpty() ? Optional.empty() : Optional.of(files);
 	}
 
+	private Optional<List<File>> decodeArchivesToShipToCluster(final Configuration configuration) {
```

Comment:
The `decodeArchivesToShipToCluster` and `decodeDirsToShipToCluster` can be merged into a single method that takes as argument the name of the config option, right?